smarts/core/utils/logging.py

```diff
@@ -101,8 +102,24 @@ def _suppress_fileout(stdname):
     except UnsupportedOperation as e:
         if not isnotebook():
             raise e
-        ## This case is notebook which does not have issues with the c_printf
-        return None
+        file = open(os.devnull, "w")
+        old_std = getattr(sys, stdname)
+        setattr(sys, stdname, file)
+
+        def cleanup(_):
+            nonlocal old_std, stdname
+            new_std = getattr(sys, stdname)
+            new_std.flush()
+            # Dummy attributes because of https://github.com/ipython/ipykernel/issues/867
+            if not hasattr(new_std, "watch_fd_thread"):
+                setattr(new_std, "watch_fd_thread", Mock())
```

Comment:
This is a clever workaround for the `AttributeError` mentioned in that issue. Does it not matter (at least for our purposes) that it isn't the true attribute? e.g. when `watch_fd_thread.join()` is called, could it cause other problems?

Reply:
I am not sure, I was tempted to just use `Mock()` in place of the file itself, but I was concerned that it would cause issues with manipulating `std` outputs. I do not think this hack matters here, it may break in the future.

On reflection, I believe I should just use `None` here.